docs/rfcs/XeGPU.md

```diff
@@ -286,7 +245,236 @@ In case that certain Xe GPU target does not support atomic operation for a certa
+```mlir
+  #sg_map_a = xegpu.sg_map<wi_layout = [1, 16], wi_data = [1, 1]>
+  %vector_a = XeGPU.load_nd %tdesc_a:
+     tensor_desc<8x16xbf16, #sg_map_b> into vector<8x1xbf16>
+```
+
+For load_nd with `packed` attribute, wi_data[0] must equal to the size required to “pack” the low-precision data into 32-bit, also known as `vnni_factor`. The result vector takes wi_data[0] as inner dimension size, to indicate the effects of layout change known as "VNNI" transformation. The data fragement <16x1xbf16> is loaded and packed as <8x1x2xbf16>. 
+
+```mlir
+  #sg_map_b = xegpu.sg_map<wi_layout = [1, 16], wi_data = [2, 1]>
+  %vector_b = XeGPU.load_nd {packed} %tdesc1:
+     tensor_desc<16x16xbf16, #sg_map_b> into vector<8x1x2xbf16>
+```
+
+For load_nd with `transpose` attribute, wi_layout is transposed to match with the tensor dimension swap. The tensor is first distributed to WI using `sg_map`, so each WI get 1x8xf32 in the example below, and then transposed to 8x1xf32. The data fragement <8x1xf32> is loaded and transposed as <1x8xf32>.
+```mlir
+  #sg_map_at = xegpu.sg_map<wi_layout = [16, 1], wi_data = [1, 1]>
+  %at = XeGPU.load_nd %tdesc1 {transpose = [1,0]} :
+     tensor_desc<16x8xf32, #sg_map_at> into vector<8x1xf32>
+```
+`xegpu.sg_map` is also applied to 1d vector load for WI data distribution. When the tensor_desc only specify 1d tensor, `sg_map.wi_layout[1]` and `sg_map.wi_data[0]` must be 1, and they are ignored in the WI distribution. 
+
+`xegpu.sg_map` is also used to describe the WI data distribution for regular load. Below example shows that each WI loads one fp32 data element. The result vector <16xfp32> is loaded and distributed to each WI as <1xf32>.
+```mlir
+  #sg_map_t = xegpu.sg_map<wi_layout = [1, 16], wi_data = [1, 1]>
+  #scatter_attr = !xegpu.tdesc_attr< memory_scope=slm, scattered=true>
+  %scatter_tdesc = XeGPU.create_tdesc, %src_addr, %offsets:
+		uint64, vector<16xindex> into tensor_desc<16xfp32, #scatter_attr, #sg_map_t>
+
+  %result = XeGPU.load_gather %scatter_tdesc, %mask {L1 = cached, L2 = uncached} :
+          tensor_desc<16xfp32, #tdesc_attr, #sg_map_t>, vector<1xi1> -> vector<1xfp32>
+```
+
+Below example shows that each WI loads 4 fp32 data element with the chunk_size_per_lane. This load with chunk_size_per_lane is effectively load 2D tensor and transpose. The data fragement <1x4xf32> is loaded and transposed as <4x1xf32>.  
+```mlir
+  #sg_map_t = xegpu.sg_map<wi_layout = [16, 1], wi_data = [1, 4]>
+  #scatter_attr = !xegpu.tdesc_attr< memory_scope=slm, scattered=true>
+  %scatter_tdesc_chunk = XeGPU.create_tdesc, %src_addr, %offsets
+		{chunk_size_per_lane=4} :
+		uint64, vector<16xindex> into tensor_desc<16x4xfp32, #scatter_attr, #sg_map_t>
+
+  %result = XeGPU.load_gather %scatter_tdesc_chunk, %mask {L1 = cached, L2 = uncached, transpose=[1,0]} :
+          tensor_desc<16x4xfp32, #tdesc_attr, #sg_map_t>, vector<1xi1> -> vector<4x1xfp32>
+```
```

Comment:
Is transposing a requirement here or just for the example's sake?

Reply:
It is a requirement.